File: src/singleton/singleton.java
// Singleton pattern

// To run this code, Java 17 or above is required.

class Singleton {
    private static Singleton singleton;

    private Singleton() {
        System.out.println("The singleton object is constructed.");
    }

    public static Singleton getInstance() {
        if (singleton == null) {
            singleton = new Singleton();
        }
        else {
            System.out.println("The singleton object is already generated.");
        }

        return singleton;
    }
}

public class Main {
    public static void main(String[] args) {
        var s1 = Singleton.getInstance();
        var s2 = Singleton.getInstance();
    }
}

File: src/singleton/singleton.cs
// Singleton pattern

// To run this code, .NET 6 or above is required.

namespace Singleton
{
    public class Singleton
    {
        private static Singleton _singleton;
        
        private Singleton()
        {
            Console.WriteLine("The singleton object is constructed.");
        }

        public static Singleton GetInstance()
        {
            if (_singleton == null)
                _singleton = new Singleton();
            else
                Console.WriteLine("The singleton object is already generated.");

            return _singleton;
        }
    }

    public static class Program
    {
        public static void Main(string[] args)
        {
            var s1 = Singleton.GetInstance();
            var s2 = Singleton.GetInstance();
        }
    }
}